### PR TITLE
refactor: split plugin into assets-sync lib and thin wasip2 wrapper

### PR DESCRIPTION
**Pull request by lwshang**
## Summary

- Extracted all sync business logic (directory scanning, MIME detection, content encoding, canister diffing) out of `plugin/` into a new `assets-sync` library crate that has no WASI/WIT dependencies and compiles natively
- `plugin/` is now a thin WASI/WIT wrapper that implements the transport layer and delegates entirely to `assets-sync`
- Introduced a `CanisterCall` trait with a single `call()` method as the boundary between the two crates, keeping the transport abstraction minimal
- Added unit tests for `sync.rs::build_operations` covering create, update, delete, no-op, type-mismatch recreate, encoding changes, and edge cases

## Test plan

- [ ] `cargo test -p assets-sync` — all `build_operations` and `scan.rs` unit tests pass natively
- [ ] `cargo build -p plugin --target wasm32-wasip2` — plugin still compiles to Wasm
- [ ] `cargo test -p ic-certified-assets` — no regressions in canister unit tests

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: assets-sync/src/sync.rs [hunk 1]
```diff
@@ -0,0 +1,571 @@
...
+
+    let canister_assets: HashMap<String, AssetDetails> = list_assets(canister)?
+        .into_iter()
+        .map(|d| (d.key.clone(), d))
+        .collect();
+    println!("canister currently has {} asset(s)", canister_assets.len());
+
+    let batch_id = create_batch(canister)?;
+    println!("created batch {batch_id}");
+
```
[the remark this answers is not in the document]
Comment (lwshang) [reply assets-sync/src/sync.rs]: Fixed in fb44fbe. `prepare_asset` now only computes metadata and stores encoded bytes; `create_batch` is deferred until after `build_operations` confirms there is actual work to do, so no-op runs never create a batch on the canister.

File: assets-sync/src/sync.rs [hunk 2]
```diff
@@ -0,0 +1,571 @@
...
+        let project = HashMap::from([mk_project_asset(
+            "/index.html",
+            "text/html",
+            &[("identity", sha.clone(), true)],
+        )]);
+        let canister = HashMap::from([mk_canister_asset(
+            "/index.html",
+            "text/html",
+            &[("identity", Some(sha))],
+        )]);
```
[the remark this answers is not in the document]
Comment (lwshang) [reply assets-sync/src/sync.rs]: Fixed in fb44fbe. Removed `everything_in_sync_returns_empty` — it was identical to `unchanged_asset_emits_no_ops` in setup and assertion.

**Review by adamspofford-dfinity — APPROVED** [review]
